src/lib.rs

```diff
@@ -637,20 +682,19 @@ impl<'text> BidiInfo<'text> {
             // Look for the start of a sequence of consecutive runs of max_level or higher.
             let mut seq_start = 0;
             while seq_start < run_count {
-                if self.levels[runs[seq_start].start] < max_level {
+                if levels[runs[seq_start].start] < max_level {
```

Comment:
We should also update the comment above the loop, which says that this loop uses the original levels instead of the ones for reordering.

Reply:
Yeah, technically it's still not "checking reordering" because in this case that vector doesn't exist here (the reorderings are managed by the runs vector), but ultimately it _is_ checking the "right" levels here unlike the other method which happens to work due to some invariants.